### ISSUE TITLE
Trace block on 1.13 and 1.14 give different result

### ISSUE THREAD

**Issue by mbratfisch-bc** [ethereum/go-ethereum#30256]
#### System information

Geth version: 1.13.15 vs 1.14.5
CL client & version: prysm
OS & Version: Linux
Commit hash : (if `develop`)

#### Expected behaviour

Traces should be the same

#### Actual behaviour

Traces are different and version 1.14.5 has a weird value field: "0xfffffffffffffffffffffffffffffffffffffffffffffffffffe9433b8211f00"

#### Steps to reproduce the behaviour

1.14.5
```
curl -X POST --data '{"jsonrpc":"2.0","method":"debug_traceBlockByHash", "params":["0x409c87178e964bf8641929ad8e58ca1a5b17afc3c83ca0b951ec353e6a99520f",  {"tracer": "callTracer"}], "id": 1001}' -H "Content-type: application/json" http://<YOUR NODE>/| jq '.result.[] | select(.txHash | contains("0x8624d132d92598cf854d9d62908222b341eed1c64ba2e5b60301282f0d4de4c8")) '
```

returns:
```
{
  "txHash": "0x8624d132d92598cf854d9d62908222b341eed1c64ba2e5b60301282f0d4de4c8",
  "result": {
    ...
    "calls": [
      {
        "from": "0xca11bde05977b3631167028862be2a173976ca11",
        "gas": "0x275e",
        "gasUsed": "0x0",
        "to": "0x105083929bf9bb22c26cb1777ec92661170d4285",
        "input": "0x",
        "value": "0x16bcc47dee100",
        "type": "CALL"
      },
      {
        "from": "0xca11bde05977b3631167028862be2a173976ca11",
        "gas": "0xa0e",
        "gasUsed": "0x0",
        "to": "0x105083929bf9bb22c26cb1777ec92661170d4285",
        "input": "0x",
        "error": "insufficient balance for transfer",
        "value": "0xfffffffffffffffffffffffffffffffffffffffffffffffffffe9433b8211f00",
        "type": "CALL"
      }
    ],
    "value": "0x0",
    "type": "CALL"
  }
}
```

1.13:
```
curl -X POST --data '{"jsonrpc":"2.0","method":"debug_traceBlockByHash", "params":["0x409c87178e964bf8641929ad8e58ca1a5b17afc3c83ca0b951ec353e6a99520f",  {"tracer": "callTracer"}], "id": 1001}' -H "Content-type: application/json" http://<YOUR NODE 1.13>/| jq '.result.[] | select(.txHash | contains("0x8624d132d92598cf854d9d62908222b341eed1c64ba2e5b60301282f0d4de4c8")) '
```

returns:
```
{
  "txHash": "0x8624d132d92598cf854d9d62908222b341eed1c64ba2e5b60301282f0d4de4c8",
  "result": {
    ...
    "calls": [
      {
        "from": "0xca11bde05977b3631167028862be2a173976ca11",
        "gas": "0x275e",
        "gasUsed": "0x0",
        "to": "0x105083929bf9bb22c26cb1777ec92661170d4285",
        "input": "0x",
        "value": "0x16bcc47dee100",
        "type": "CALL"
      }
    ],
    "value": "0x0",
    "type": "CALL"
  }
}
```

#### Backtrace

````
[backtrace]
````

When submitting logs: please submit them as text and not screenshots.

**Comment by easyfold** [comment]
@mbratfisch-bc I think the different is due to a "bug" that is fixed in v1.14

You can see code in v1.13, `ErrInsufficientBalance` in subcall is not captured by tracer.
https://github.com/ethereum/go-ethereum/blob/c5ba367eb6232e3eddd7d6226bfd374449c63164/core/vm/evm.go#L179-L191

And it's fixed in v1.14.
https://github.com/ethereum/go-ethereum/blob/aa55f5ea200dfd07618fdf658d9d2741c3b376a8/core/vm/evm.go#L190-L205

Also, two subcalls show in [explorer](https://etherscan.io/tx/0x8624d132d92598cf854d9d62908222b341eed1c64ba2e5b60301282f0d4de4c8#internal)

**Comment by holiman** [comment]
Ideally, things never change. In reality, they sometimes do, even though we do our best to prevent it. 

The current code will show the failed call. Whether that is desireable or not can be debated, but at least it is not Wrong. Arguably, the consumer can always filter and remove calls which has an error on them, and regain the first view. 

I'll close this, since it's not a bug to fix per se.
